File: fluxo-pessoal/backend/app/models/reserve_box.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import Boolean, DateTime, ForeignKey, Numeric, String, Text, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class ReserveBox(Base):
    __tablename__ = "reserve_boxes"
    __table_args__ = (UniqueConstraint("account_id", "name", name="uq_reserve_boxes_account_name"),)

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    account_id: Mapped[int] = mapped_column(ForeignKey("accounts.id"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    current_balance: Mapped[Decimal] = mapped_column(Numeric(14, 2), nullable=False, default=0)
    target_amount: Mapped[Decimal | None] = mapped_column(Numeric(14, 2))
    notes: Mapped[str | None] = mapped_column(Text)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    account = relationship("Account", back_populates="reserve_boxes")

File: frontend/src/types/reserveBox.ts
export interface ReserveBox {
  id: number;
  account_id: number;
  chart_account_id: number | null;
  chart_account_code: string | null;
  chart_account_name: string | null;
  withdrawal_chart_account_id: number | null;
  withdrawal_chart_account_code: string | null;
  withdrawal_chart_account_name: string | null;
  name: string;
  current_balance: string;
  target_amount: string | null;
  notes: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}
